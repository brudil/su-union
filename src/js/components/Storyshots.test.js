import initStoryshots from '@storybook/addon-storyshots';

/* TODO: add to each set before setup - to check for proptype warnings
console.error = jest.fn();
expect(console.error.mock.calls.length).toBe(0);
*/

initStoryshots();
