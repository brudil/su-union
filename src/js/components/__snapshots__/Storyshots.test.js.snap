// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots BackBar supports brand color 1`] = `
<div>
  <div
    className="BackBar BackBar--color-red"
  >
    <a
      href={undefined}
    >
      <svg
        className="BackBar__arrow"
        height="18"
        viewBox="0 0 18 18"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fillRule="evenodd"
        >
          <rect
            className="BackBar__arrow-back"
            height="18"
            rx="2"
            width="18"
          />
          <path
            className="BackBar__arrow"
            d="M15.715215,9.67002437 C16.0969496,9.29253227 16.0939089,8.67853807 15.7121731,8.30258285 L9.55867891,2.24158095 C8.82411406,1.50934734 7.72454442,2.60694484 8.45759085,3.33917457 L13.0809849,7.94634327 C13.2330683,8.09946147 13.1828826,8.22377469 12.9684427,8.22377469 L2.77867689,8.22377469 C2.34826324,8.22377469 2,8.57094432 2,8.99998035 C2,9.42901639 2.34827492,9.77618601 2.77867689,9.77618601 L12.9684427,9.77618601 C13.1844049,9.77618601 13.2361129,9.89746815 13.0809849,10.0521038 L8.45759085,14.660825 C7.72454442,15.3930586 8.82411795,16.4906522 9.55867891,15.7584186 L15.715215,9.67002437 Z"
            fill="#FFF"
            fillRule="nonzero"
            transform="matrix(-1 0 0 1 18 0)"
          />
        </g>
      </svg>
      Support
    </a>
  </div>
  <div
    className="BackBar BackBar--color-green"
  >
    <a
      href={undefined}
    >
      <svg
        className="BackBar__arrow"
        height="18"
        viewBox="0 0 18 18"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fillRule="evenodd"
        >
          <rect
            className="BackBar__arrow-back"
            height="18"
            rx="2"
            width="18"
          />
          <path
            className="BackBar__arrow"
            d="M15.715215,9.67002437 C16.0969496,9.29253227 16.0939089,8.67853807 15.7121731,8.30258285 L9.55867891,2.24158095 C8.82411406,1.50934734 7.72454442,2.60694484 8.45759085,3.33917457 L13.0809849,7.94634327 C13.2330683,8.09946147 13.1828826,8.22377469 12.9684427,8.22377469 L2.77867689,8.22377469 C2.34826324,8.22377469 2,8.57094432 2,8.99998035 C2,9.42901639 2.34827492,9.77618601 2.77867689,9.77618601 L12.9684427,9.77618601 C13.1844049,9.77618601 13.2361129,9.89746815 13.0809849,10.0521038 L8.45759085,14.660825 C7.72454442,15.3930586 8.82411795,16.4906522 9.55867891,15.7584186 L15.715215,9.67002437 Z"
            fill="#FFF"
            fillRule="nonzero"
            transform="matrix(-1 0 0 1 18 0)"
          />
        </g>
      </svg>
      What's on
    </a>
  </div>
  <div
    className="BackBar BackBar--color-blue"
  >
    <a
      href={undefined}
    >
      <svg
        className="BackBar__arrow"
        height="18"
        viewBox="0 0 18 18"
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fillRule="evenodd"
        >
          <rect
            className="BackBar__arrow-back"
            height="18"
            rx="2"
            width="18"
          />
          <path
            className="BackBar__arrow"
            d="M15.715215,9.67002437 C16.0969496,9.29253227 16.0939089,8.67853807 15.7121731,8.30258285 L9.55867891,2.24158095 C8.82411406,1.50934734 7.72454442,2.60694484 8.45759085,3.33917457 L13.0809849,7.94634327 C13.2330683,8.09946147 13.1828826,8.22377469 12.9684427,8.22377469 L2.77867689,8.22377469 C2.34826324,8.22377469 2,8.57094432 2,8.99998035 C2,9.42901639 2.34827492,9.77618601 2.77867689,9.77618601 L12.9684427,9.77618601 C13.1844049,9.77618601 13.2361129,9.89746815 13.0809849,10.0521038 L8.45759085,14.660825 C7.72454442,15.3930586 8.82411795,16.4906522 9.55867891,15.7584186 L15.715215,9.67002437 Z"
            fill="#FFF"
            fillRule="nonzero"
            transform="matrix(-1 0 0 1 18 0)"
          />
        </g>
      </svg>
      Latest news
    </a>
  </div>
</div>
`;

exports[`Storyshots BackBar with text 1`] = `
<div
  className="BackBar BackBar--color-blue"
>
  <a
    href={undefined}
  >
    <svg
      className="BackBar__arrow"
      height="18"
      viewBox="0 0 18 18"
      width="18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fillRule="evenodd"
      >
        <rect
          className="BackBar__arrow-back"
          height="18"
          rx="2"
          width="18"
        />
        <path
          className="BackBar__arrow"
          d="M15.715215,9.67002437 C16.0969496,9.29253227 16.0939089,8.67853807 15.7121731,8.30258285 L9.55867891,2.24158095 C8.82411406,1.50934734 7.72454442,2.60694484 8.45759085,3.33917457 L13.0809849,7.94634327 C13.2330683,8.09946147 13.1828826,8.22377469 12.9684427,8.22377469 L2.77867689,8.22377469 C2.34826324,8.22377469 2,8.57094432 2,8.99998035 C2,9.42901639 2.34827492,9.77618601 2.77867689,9.77618601 L12.9684427,9.77618601 C13.1844049,9.77618601 13.2361129,9.89746815 13.0809849,10.0521038 L8.45759085,14.660825 C7.72454442,15.3930586 8.82411795,16.4906522 9.55867891,15.7584186 L15.715215,9.67002437 Z"
          fill="#FFF"
          fillRule="nonzero"
          transform="matrix(-1 0 0 1 18 0)"
        />
      </g>
    </svg>
    Hello Button
  </a>
</div>
`;

exports[`Storyshots Button default 1`] = `
<div
  className="Container"
>
  <a
    className="Button"
    href="https://maps.google.com"
  >
    Find us on Google Maps
  </a>
</div>
`;

exports[`Storyshots ContentCard with text 1`] = `
<div
  className="ContentCard"
>
  Hello Content!
</div>
`;

exports[`Storyshots ContentNavigation default 1`] = `
<div
  className="NavigationCard"
>
  <h3
    className="NavigationCard__title"
  >
    Navigation
  </h3>
  <ul
    className="NavigationCard__list"
  >
    <li
      className="NavigationCard__item"
    >
      <a
        className="NavigationCard__anchor"
        href="#animals"
      >
        Animals
      </a>
    </li>
    <li
      className="NavigationCard__item NavigationCard__item--active"
    >
      <a
        className="NavigationCard__anchor"
        href="#boats"
      >
        Boats
      </a>
    </li>
    <li
      className="NavigationCard__item"
    >
      <a
        className="NavigationCard__anchor"
        href="#cars"
      >
        Cars
      </a>
    </li>
  </ul>
</div>
`;

exports[`Storyshots EventsCalender standard event card 1`] = `
<div
  className="Container"
>
  <div
    className="EventsCalender__item"
  >
    <a
      className="u-faux-link"
      href="/ents/event/2791/"
    />
    <div
      className="EventsCalender__item-image u-responsive-ratio u-responsive-ratio--wide"
    >
      <div
        className="TEST_IMAGE"
        data-props="original_images/67e36f330356451daed64e4011d25616"
      />
    </div>
    <div
      className="EventsCalender__item-container"
    >
      <h2
        className="EventsCalender__item-title"
      >
        Introduction to the Role Models Project
      </h2>
      <div
        className="EventsCalender__item-description"
      >
        Join us to find out more about volunteering with the Role Models Project! We will be giving an introduction to what the project is all about and what to expect as a new volunteer.
      </div>
      <div
        className="EventsCalender__item-meta"
      >
        6:00pm
        <span>
           - 
        </span>
        8:00pm
        <span>
           / 
        </span>
        Fulton 104
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots FigureCollection default 1`] = `
<div
  className="Container"
>
  <ul
    className="FigureCollection"
  >
    <li
      className="FigureCollection__item"
    >
      <a
        className="FigureCollection__link"
        href="/about-us/full-time-elected-officers/activities"
      >
        <div
          className="u-responsive-ratio u-responsive-ratio--square"
        >
          <div
            className="TEST_IMAGE"
            data-props="images/4759cccd3b3f4ed4813a260cc0d4833f.original.jpg"
          />
        </div>
        <span
          className="FigureCollection__title"
        >
          Lucy Williams
        </span>
        <span
          className="FigureCollection__secondary"
        >
          Activities Officer
        </span>
      </a>
    </li>
    <li
      className="FigureCollection__item"
    >
      <a
        className="FigureCollection__link"
        href="/about-us/full-time-elected-officers/society-citizenship"
      >
        <div
          className="u-responsive-ratio u-responsive-ratio--square"
        >
          <div
            className="TEST_IMAGE"
            data-props="images/8da8156f2e0e40b4878dcbea569ce2b5.original.jpg"
          />
        </div>
        <span
          className="FigureCollection__title"
        >
          Aisling Murray
        </span>
        <span
          className="FigureCollection__secondary"
        >
          Society and Citizenship Officer
        </span>
      </a>
    </li>
    <li
      className="FigureCollection__item"
    >
      <a
        className="FigureCollection__link"
        href="/about-us/full-time-elected-officers/undergraduate-education"
      >
        <div
          className="u-responsive-ratio u-responsive-ratio--square"
        >
          <div
            className="TEST_IMAGE"
            data-props="images/241b436e49524d8da68faf1f1f83c4b6.original.jpg"
          />
        </div>
        <span
          className="FigureCollection__title"
        >
          Lulah Stratheran Brady
        </span>
        <span
          className="FigureCollection__secondary"
        >
          Undergraduate Education Officer
        </span>
      </a>
    </li>
    <li
      className="FigureCollection__item"
    >
      <a
        className="FigureCollection__link"
        href="/about-us/full-time-elected-officers/postgraduate-education"
      >
        <div
          className="u-responsive-ratio u-responsive-ratio--square"
        >
          <div
            className="TEST_IMAGE"
            data-props="images/f07caf2bb1114cb8a18dd49a6e9e1045.original.jpg"
          />
        </div>
        <span
          className="FigureCollection__title"
        >
          Sarah McIntosh
        </span>
        <span
          className="FigureCollection__secondary"
        >
          Postgraduate Education Officer
        </span>
      </a>
    </li>
    <li
      className="FigureCollection__item"
    >
      <a
        className="FigureCollection__link"
        href="/about-us/full-time-elected-officers/president"
      >
        <div
          className="u-responsive-ratio u-responsive-ratio--square"
        >
          <div
            className="TEST_IMAGE"
            data-props="images/16664de397694b5ea5e194b0a1fa73dc.original.jpg"
          />
        </div>
        <span
          className="FigureCollection__title"
        >
          Frida Gustafsson
        </span>
        <span
          className="FigureCollection__secondary"
        >
          President
        </span>
      </a>
    </li>
    <li
      className="FigureCollection__item"
    >
      <a
        className="FigureCollection__link"
        href="/about-us/full-time-elected-officers/welfare"
      >
        <div
          className="u-responsive-ratio u-responsive-ratio--square"
        >
          <div
            className="TEST_IMAGE"
            data-props="images/d05f369bcabc47ed83ecaac0e36a9adb.original.jpg"
          />
        </div>
        <span
          className="FigureCollection__title"
        >
          Grainne Gahan
        </span>
        <span
          className="FigureCollection__secondary"
        >
          Welfare Officer
        </span>
      </a>
    </li>
  </ul>
</div>
`;

exports[`Storyshots HeadingHero title only 1`] = `
<div
  className="Container"
>
  <div
    className="TEST_IMAGE"
    data-props="/images/4219b2966c1047dd8fe4bfa2aa922c72.original.jpg"
  >
    <h1
      className="HeadingImage__title"
    >
      Our holiday opening times
    </h1>
  </div>
</div>
`;

exports[`Storyshots HeadingHero width description 1`] = `
<div
  className="Container"
>
  <div
    className="TEST_IMAGE"
    data-props="/images/4219b2966c1047dd8fe4bfa2aa922c72.original.jpg"
  >
    <h1
      className="HeadingImage__title"
    >
      Beware of seagulls
    </h1>
    <div>
      <div
        className="HeadingImage__desc"
      >
        They are everywhere. They will get you.
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Loader dark 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#333333",
      "height": "100%",
      "width": "100%",
    }
  }
>
  <div
    className="RippleLoader"
  >
    <div
      style={
        Object {
          "borderColor": "#fff",
        }
      }
    />
  </div>
</div>
`;

exports[`Storyshots Loader light 1`] = `
<div
  className="RippleLoader"
>
  <div
    style={
      Object {
        "borderColor": "#000",
      }
    }
  />
</div>
`;

exports[`Storyshots NewsBlock with images 1`] = `
<ul
  className="NewsGrid NewsGrid--full-width"
>
  <li
    className="NewsGrid__item NewsBlock"
  >
    <a
      className="NewsBlock__link"
      href="https://example.com"
    >
      <div
        className="NewsBlock__image"
      >
        <div
          className="u-responsive-ratio u-responsive-ratio--43"
        >
          <img
            alt=""
            src="https://www.sussexstudent.com/asset/News/6013/Lucy.jpg"
          />
        </div>
      </div>
      <div
        className="NewsBlock__content"
      >
        <div
          className="NewsBlock__title-link"
        >
          <h2
            className="NewsBlock__title"
          >
            Union launch week of meta headlines for styleguide
          </h2>
        </div>
        <p
          className="NewsBlock__standfirst"
        >
          A small sentence or two about the news article
        </p>
        <div
          className="NewsBlock__meta"
        >
          <date
            className="NewsBlock__date"
          >
            Today
          </date>
        </div>
      </div>
    </a>
  </li>
</ul>
`;

exports[`Storyshots NewsBlock without image 1`] = `
<ul
  className="NewsGrid NewsGrid--full-width"
>
  <li
    className="NewsGrid__item NewsBlock"
  >
    <a
      className="NewsBlock__link"
      href="https://example.com"
    >
      <div
        className="NewsBlock__image"
      >
        <div
          className="u-responsive-ratio u-responsive-ratio--43"
        >
          <div
            className="NewsBlock__image--default"
          >
            <svg
              height="841.89"
              viewBox="0 0 595.28 841.89"
              width="595.28"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="#27428c"
              >
                <path
                  d="M581.94 434.428c-.874 5.655-2.931 13.934-4.677 18.542l4.438-.478c2.227-3.658 5.658-12.246 7.471-18.854l-7.232.79zm-40.079 51.615c2.868 1.914 8.515 3.498 13.685 3.498 12.648 0 18.764-7.24 18.764-15.508 0-7.876-4.615-12.251-13.666-15.745-7.405-2.859-10.664-5.318-10.664-10.327 0-3.659 2.783-8.04 10.096-8.04a21.51 21.51 0 0 1 10.188 2.546l1.904-5.642c-2.398-1.35-6.372-2.622-11.854-2.622-10.414 0-17.329 6.194-17.329 14.546 0 7.552 5.409 12.086 14.149 15.198 7.238 2.78 10.095 5.64 10.095 10.647 0 5.396-4.126 9.138-11.207 9.138-4.772 0-9.302-1.593-12.414-3.491l-1.747 5.802zm-23.939 2.613h7.008v-47.699h16.394v-5.882H501.63v5.882h16.292v47.699zm-53.739 0v-22.888c0-8.917-.147-15.348-.549-22.102l.249-.086c2.687 5.805 6.27 11.924 9.998 17.895l17.024 27.181h6.996v-53.581h-6.525v22.426c0 8.342.176 14.859.8 22.014l-.154.089c-2.551-5.48-5.65-11.141-9.625-17.418l-17.177-27.11h-7.557v53.581h6.52zm-15.663-30.842h-20.827v-16.943h22.03v-5.796h-28.945v53.581h30.129v-5.796h-23.214v-19.316h20.827v-5.73zm-71.556-16.943c1.836-.388 4.534-.711 8.105-.711 14.632 0 21.792 8.026 21.724 20.675 0 14.476-8.039 22.736-22.828 22.736-2.705 0-5.25-.07-7.001-.396v-42.304zm-6.91 47.635c3.491.399 7.623.638 12.555.638 10.414 0 18.679-2.71 23.708-7.709 4.999-4.932 7.69-12.095 7.69-20.838 0-8.675-2.771-14.87-7.538-19.25-4.705-4.357-11.773-6.672-21.717-6.672-5.475 0-10.486.478-14.698 1.111v52.72zm-49.289-53.431v31.554c0 16.635 8.258 22.912 19.228 22.912 11.615 0 20.288-6.767 20.288-23.214v-31.252h-7.013v31.728c0 11.839-5.085 17.095-13.038 17.095-7.147 0-12.477-5.094-12.477-17.095v-31.728h-6.988zm-27.047 53.581h7.001v-47.699h16.371v-5.882h-39.676v5.882h16.304v47.699zm-49.852-2.613c2.857 1.914 8.51 3.498 13.673 3.498 12.647 0 18.77-7.24 18.77-15.508 0-7.876-4.621-12.251-13.679-15.745-7.398-2.859-10.658-5.318-10.658-10.327 0-3.659 2.783-8.04 10.102-8.04 4.853 0 8.424 1.598 10.175 2.546l1.904-5.642c-2.387-1.35-6.354-2.622-11.848-2.622-10.407 0-17.323 6.194-17.323 14.546 0 7.552 5.396 12.086 14.143 15.198 7.239 2.78 10.095 5.64 10.095 10.647 0 5.396-4.132 9.138-11.2 9.138-4.773 0-9.309-1.593-12.409-3.491l-1.745 5.802zM442.88 557.255v-13.21c0-8.661-.152-16.775-.47-24.005h.231c2.545 6.518 6.043 13.672 9.07 19.403l9.064 17.812h15.101v-53.6H462.76v12.635c0 8.043.324 15.512 1.123 22.58h-.249c-2.388-6.27-5.487-13.438-8.431-18.755l-8.502-16.46h-16.945v53.6h13.124zm-44.999-43.097c6.909 0 11.049 6.992 11.049 16.218 0 9.616-4.06 16.38-10.975 16.38-6.916 0-11.298-6.202-11.298-16.228 0-9.851 4.382-16.37 11.224-16.37m.232-11.375c-16.377 0-26.632 12.403-26.632 27.991 0 14.864 9.126 27.348 25.748 27.348 16.559 0 26.887-10.976 26.887-28.229 0-14.31-8.741-27.11-26.003-27.11m-46.671 54.472h14.381v-53.6h-14.381v53.6zm-40.958 0v-13.21c0-8.661-.146-16.775-.476-24.005h.238c2.545 6.518 6.056 13.672 9.064 19.403l9.064 17.812h15.113v-53.6h-13.124v12.635c0 8.043.323 15.512 1.118 22.58h-.232c-2.388-6.27-5.481-13.438-8.43-18.755l-8.515-16.46h-16.939v53.6h13.119zm-66.618-53.6v29.185c0 17.255 7.716 25.282 22.414 25.282 15.04 0 23.305-8.351 23.305-24.887v-29.58h-14.394v30.38c0 8.497-2.784 12.721-8.509 12.721-5.646 0-8.429-4.614-8.429-12.721v-30.38h-14.387zM5.538 244.874S.13 347.988 40.623 406.024c40.488 58.027 105.818 68.386 186.799 72.078l.263-105.369s-2.972-72.253-94.471-107.619c0 0-44.003-20.24-127.676-20.24M350.521 308.575s2.538 48.333-16.444 75.541c-18.971 27.197-49.601 32.056-87.561 33.79l-.123-49.406s1.392-33.85 44.284-50.428c.001-.001 20.632-9.497 59.844-9.497"
                />
              </g>
            </svg>
          </div>
        </div>
      </div>
      <div
        className="NewsBlock__content"
      >
        <div
          className="NewsBlock__title-link"
        >
          <h2
            className="NewsBlock__title"
          >
            Union launch week of meta headlines for styleguide
          </h2>
        </div>
        <p
          className="NewsBlock__standfirst"
        >
          A small sentence or two about the news article
        </p>
        <div
          className="NewsBlock__meta"
        >
          <date
            className="NewsBlock__date"
          >
            Today
          </date>
        </div>
      </div>
    </a>
  </li>
</ul>
`;
